neo/rawio/neuralynxrawio/neuralynxrawio.py

```diff
@@ -128,32 +131,68 @@ class NeuralynxRawIO(BaseRawIO):
         ("channel_id", "uint32"),
         ("sample_rate", "uint32"),
         ("nb_valid", "uint32"),
-        ("samples", "int16", (NcsSection._RECORD_SIZE)),
+        ("samples", "int16", NcsSection._RECORD_SIZE),
     ]
 
     def __init__(
-        self, dirname="", filename="", exclude_filename=None, keep_original_times=False, strict_gap_mode=True, **kargs
+            self,
+            dirname="",
+            include_filenames=None,
+            exclude_filenames=None,
+            keep_original_times=False,
+            strict_gap_mode=True,
+            filename=None,
+            exclude_filename=None,
+            **kargs
     ):
 
-        if dirname != "":
-            self.dirname = dirname
-            self.rawmode = "one-dir"
-        elif filename != "":
-            self.filename = filename
-            self.rawmode = "one-file"
+        if not dirname:
+            raise ValueError("`dirname` cannot be empty.")
+
+        if filename:
+            include_filenames = filename
+            warnings.warn("`filename` is deprecated and will be removed. Please use `include_filenames` instead")
+
+        if exclude_filename:
+            exclude_filenames = exclude_filename
+            warnings.warn("`exclude_filename` is deprecated and will be removed. Please use `exclude_filenames` instead")
+
+        if include_filenames is None:
+            include_filenames = []
+        elif not isinstance(include_filenames, (list, set, np.ndarray)):
+            include_filenames = [include_filenames]
+
+        if exclude_filenames is None:
+            exclude_filenames = set()
```

Comment:
Why are we using a set? Are you worried that the user will put in the same file twice? That seems like a user error that we don't need to fix.

Reply:
This is minor, but a set is faster to check if an element is in it than a list. Do you prefer to use list here?

Reply:
Are you sure? At what scale?
```python
%timeit "a" in {"a", "b", "c"}
23.5 ns ± 0.245 ns per loop (mean ± std. dev. of 7 runs, 10,000,000 loops each)

%timeit "a" in ["a", "b", "c"]
20.3 ns ± 0.201 ns per loop (mean ± std. dev. of 7 runs, 10,000,000 loops each)
```

Reply:
In Python, a set is implemented with a hashtable, whereas a list is a dynamic array. The time complexity of membership testing should be O(1) and O(n), respectively. The difference is negligible when the list size is small (as in your test above). Here is a related post on StackOverflow: https://stackoverflow.com/questions/8929284/what-makes-sets-faster-than-lists.

I am open to both list and set as we may not have a huge number of files to read.

Reply:
I think that is a good point for speed in the case of a large list of files. I'm not sure of the limit for neuralynx files in a folder but I would guess that it is <10 or 15 files (correct me if I'm wrong). Not to mention even if I increase the size of items to 7 for set it was 30 ns for each item and for list the last item took 70 ns (and 20ns for the first) so for our purpose of running this once it is pretty negligible. I like a list better unless I'm explicitly forcing items to only be included once and hopefully if someone really wants to exclude a lot of items they instead switch to include instead (which you proposed ! )